### PR TITLE
8339336: Fix build system whitespace to adhere to coding conventions

### PR DESCRIPTION
**Pull request by magicus**
The build system code has unfortunately diverted in some places from the conventions as described in https://openjdk.org/groups/build/doc/code-conventions.html.

Instead of trying to fix these when touching code nearby, I'd like to make an effort to fix all issues at once and separately. Incremental fixes has their benefit, but they can also muddy the actual fix and are not always appreciated.

The updates in this patch have all been discovered using automated tools, but each and every change has been manually scrutinized. Those that the automatic tools pointed out that, but that were not obviously or clear-cut safe (e.g. adding spaces after comma, in `subst` or similar situations) were reverted before I pushed. I chose to err on the "First, do no harm" side, so there might be places that could have been corrected, but were not. 

I have made a single type of change per commit in this branch. It might be easier to review this by looking at one commit at a time.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339336](https://bugs.openjdk.org/browse/JDK-8339336): Fix build system whitespace to adhere to coding conventions (**Bug** - P3)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20798/head:pull/20798` \
`$ git checkout pull/20798`

Update a local copy of the PR: \
`$ git checkout pull/20798` \
`$ git pull https://git.openjdk.org/jdk.git pull/20798/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20798`

View PR using the GUI difftool: \
`$ git pr show -t 20798`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20798.diff">https://git.openjdk.org/jdk/pull/20798.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20798#issuecomment-2321936245)

### CODE REVIEW

File: make/test/BuildMicrobenchmark.gmk [hunk 1]
```diff
@@ -67,8 +67,8 @@ MICROBENCHMARK_NATIVE_EXCLUDE :=
 ###
 
 # Need double \n to get new lines and no trailing spaces
-MICROBENCHMARK_MANIFEST := Build: $(FULL_VERSION)\n\
-\nJMH-Version: $(JMH_VERSION)\n\
+MICROBENCHMARK_MANIFEST := Build: $(FULL_VERSION)\n \
+\nJMH-Version: $(JMH_VERSION)\n \
```
Comment (erikj79) [inline make/test/BuildMicrobenchmark.gmk:71]: Are you sure these spaces are safe? The comment mentions trailing spaces as something undesired and depending on how this is used, I suspect adding a space between the \n could interfere with that.

**Review by erikj79 — COMMENTED** [review]
I'm pretty sure most of these are safe changes, but there are some where I'm a bit hesitant. I would feel safer if you ran a full compare build on it and also manually verified a few incremental build scenarios, including running tests.

In general I very much approve of this patch, thank you for taking the time to clean house! I'm amazed at the number of 79 character wide comment line dividers.

File: make/test/BuildMicrobenchmark.gmk (hunk 1 above)
Comment (magicus) [reply make/test/BuildMicrobenchmark.gmk:71]: Now that I look at this again, I realize the broken lines start at position 1. That indicates that the original author did not want to indent the line as it "should" have been, just to make sure that no extra space were added. So you are likely correct; this was a bad change. Thanks for spotting it.

File: make/MainSupport.gmk [hunk 2]
```diff
@@ -197,8 +197,8 @@ endef
 # $1_MODULES : All modules that had rules generated
 # $1_TARGETS : All targets generated
 define DeclareRecipesForPhase
-  $(foreach i,2 3 4 5 6 7 8, $(if $(strip $($i)),$(strip $1)_$(strip $($i)))$(NEWLINE))
-  $(if $(9),$(error Internal makefile error: Too many arguments to \
+  $(foreach i, 2 3 4 5 6 7 8, $(if $(strip $($i)),$(strip $1)_$(strip $($i)))$(NEWLINE))
+  $(if $(9), $(error Internal makefile error: Too many arguments to \
       DeclareRecipesForPhase, please update MakeHelper.gmk))
```
[the remark this answers is not in the document]
Comment (magicus) [reply make/MainSupport.gmk:202]: Yes, I agree on all accounts. We have some other places left as well where `NamedParamsMacroTemplate` is not used even though it should be. Feel free to open a bug.

**Comment by magicus** [comment]
Running compare builds sounds like a good safety measure. 

I've done that now, and it looks clean on all platforms. (With one exception: we include the vardeps files for jar files in jdk/demos -- historical reasons, I assume -- and they look slightly different; but they should not really have been included in the comparison anyway so it does not matter.)

File: make/MainSupport.gmk (hunk 2 above)
[the remark this answers is not in the document]
Comment (magicus) [reply make/MainSupport.gmk:202]: I opened https://bugs.openjdk.org/browse/JDK-8339370

**Comment by magicus** [comment]
> also manually verified a few incremental build scenarios, including running tests.

I have tested changing:
* a hotspot file
* a JDK library native file
* a JDK module java file

and verified that the correct bits get rebuilt. I have also verified that running tests work, including after all these incremental rebuilds.

**Pull request closed without merging**